bundle/src/main/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionJob.java

```diff
@@ -0,0 +1,126 @@
+        Integer maxAgeInDays = job.getProperty(MAX_AGE_IN_DAYS, Integer.class);
+        LOG.debug("Job Configuration: ["
+                + "Group Name: {}, "
+                + "Service User: {}, "
+                + "Age of Package {} days,]", groupName, SERVICE_USER, maxAgeInDays);
+
+        try (ResourceResolver resourceResolver = resourceResolverFactory.getServiceResourceResolver(
+            Collections.singletonMap(ResourceResolverFactory.SUBSERVICE, SERVICE_USER))) {
+            session = resourceResolver.adaptTo(Session.class);
+            JcrPackageManager packageManager = packaging.getPackageManager(session);
+            List<JcrPackage> packages = packageManager.listPackages(groupName, false);
+
+            for (JcrPackage jcrPackage : packages) {
+                String packageDescription = getPackageDescription(jcrPackage);
+                if (isPackageOldEnough(jcrPackage, maxAgeInDays)) {
+                    packageManager.remove(jcrPackage);
+                    LOG.info("Deleted package {}", packageDescription);
+                } else {
+                    LOG.info("Not removing package because it's not old enough {}", packageDescription);
+                }
+            }
+        } catch (LoginException | RepositoryException | IOException e) {
+            LOG.error("Unable to clear packages due to {}", e.toString());
+            return JobResult.FAILED;
+        } finally {
+            if (session != null) {
+                session.logout();
+            }
+        }
+        LOG.info("Package Garbage Collector job finished");
+        return JobResult.OK;
+    }
+
+    private boolean isPackageOldEnough(JcrPackage jcrPackage, Integer maxAgeInDays) throws RepositoryException, IOException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM);
+        Period maxAge = Period.ofDays(maxAgeInDays);
+        LocalDate oldestAge = LocalDate.now().minus(maxAge);
+        Calendar packageCreatedAtCalendar = jcrPackage.getPackage().getCreated();
+        LocalDate packageCreatedAt = LocalDateTime.ofInstant(
+            packageCreatedAtCalendar.toInstant(),
+            packageCreatedAtCalendar.getTimeZone().toZoneId()).toLocalDate();
+        String packageDescription = getPackageDescription(jcrPackage);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Checking if package is old enough: Name: {}, Created At: {}, Oldest Age: {}",
+                packageDescription, packageCreatedAt.format(formatter), oldestAge.format(formatter));
+        }
+        return packageCreatedAt.isBefore(oldestAge) || packageCreatedAt.isEqual(oldestAge);
+    }
+
+    private String getPackageDescription(JcrPackage jcrPackage) throws RepositoryException {
+        Node packageNode = jcrPackage.getNode();
+        if (packageNode != null) {
+            return packageNode.getPath();
+        }
+        return "Unknown package";
```

Comment:
I don't think that the code receiving this message is capable to handle this return string... Can you return ```null``` and let the invoker check the return value?

Reply:
This is just used by the log messages - and will only happen if there is an issue accessing the meta data from the repository.